test/testdata/desugar/andand_pinned.rb

```diff
@@ -0,0 +1,16 @@
+# typed: strict
+extend T::Sig
+
+sig {returns(T::Boolean)}
```

Comment:
Can you add a test to show that we still emit a pinning error if this method’s return type did widen the type?

And because I’m curious, would you mind another test that demonstrates what happens when `x` starts out untyped? Not sure if the behavior on that example changes in this PR but `¯\_(ツ)_/¯` maybe it’ll be a good test to have